src/Lecture4.hs

```diff
@@ -183,8 +233,28 @@ The 'Stats' data type has multiple fields. All these fields have
 instance for the 'Stats' type itself.
 -}
 
-instance Semigroup Stats where
+-- define a strict version of <> similar to $!
+(<>!) :: (Semigroup a) => a -> a -> a
+(<>!)  !a !b = let !x = a <> b in x
+infixr 6 <>!
 
+sappendMaybe' :: (Semigroup a) => Maybe a -> Maybe a -> Maybe a
+sappendMaybe' (Just !a) (Just !b) = Just $ a <>! b
+sappendMaybe' a b = a <>! b
```

Comment:
That's absolutely correct implementation 👍🏻 
You can write it a bit shorter though by pattern matching only on the final result:

```suggestion
sappendMaybe' a b = case a <> b of
    Nothing -> Nothing
    Just !x -> Just x
```